src/app/wallet-ui-get/wallet-ui-get.component.ts

```diff
@@ -0,0 +1,195 @@
+import { AfterViewInit, Component, OnInit, TemplateRef, ViewChild } from '@angular/core';
+import { ActivatedRoute, NavigationExtras, Router } from '@angular/router';
+import { NgbModal, NgbModalOptions, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { forkJoin, Observable, of } from 'rxjs';
+import { delay, concatMap, map, switchMap } from 'rxjs/operators';
+import { AuthService } from 'src/app/services/auth/auth.service';
+import { CredentialService } from 'src/app/services/credential/credential.service';
+import { DataService } from 'src/app/services/data/data-request.service';
+import { GeneralService } from 'src/app/services/general/general.service';
+import { IImpressionEventInput } from 'src/app/services/telemetry/telemetry-interface';
+import { TelemetryService } from 'src/app/services/telemetry/telemetry.service';
+import { ToastMessageService } from 'src/app/services/toast-message/toast-message.service';
+import * as CredentialHandlerPolyfill from 'credential-handler-polyfill';
+import * as WebCredentialHandler from 'web-credential-handler';
+@Component({
+  selector: 'app-browse-documents',
+  templateUrl: './wallet-ui-get.component.html',
+  styleUrls: ['./wallet-ui-get.component.scss']
+})
+export class WalletUiGetComponent implements OnInit, AfterViewInit {
+  categories = [];
+  isLoading = false;
+  approvalModalRef: NgbModalRef;
+  showApproval = false;
+  isClaimRejected = false;
+  allCredentials: any;
+  showCredentialList = false;
+  selectedCategory: string = '';
+  @ViewChild('approvalModal') approvalModal: TemplateRef<any>;
+
+  constructor(
+    private router: Router,
+    public generalService: GeneralService,
+    public authService: AuthService,
+    private dataService: DataService,
+    private toastMessage: ToastMessageService,
+    private activatedRoute: ActivatedRoute,
+    private readonly modalService: NgbModal,
+    private telemetryService: TelemetryService,
+    private readonly credentialService: CredentialService
+  ) { }
+
+  ngOnInit(): void {
+    if (this.credentialService.selectedCategory) {
+      this.selectedCategory = this.credentialService.selectedCategory;
+      this.showCredentialList = true;
+      this.allCredentials = [...this.credentialService.credentialList];
+    } else {
+      this.fetchCredentialCategories();
```

Comment:
@ak-ironix Keep only required function (handleEvent) rest if not required remove